pkg/policy/rule.go

```diff
@@ -204,5 +215,55 @@ func (r *rule) canReach(ctx *SearchContext, state *traceState) api.Decision {
 		}
 	}
 
+	return entitiesDecision
+}
+
+func (r *rule) canReachEntities(ctx *SearchContext, state *traceState) api.Decision {
+	entitiesFound := false
+	for _, r := range r.Egress {
+		for _, entity := range r.ToEntities {
+			entitiesFound = true
+			l, ok := api.EntitySelectorMapping[entity]
+			if !ok {
+				ctx.PolicyTrace("     unsupported entity found: %s, skipping", entity)
+				continue
+			}
+
+			if l.Matches(ctx.To) {
+				ctx.PolicyTrace("+     Found all required labels to match entity %s\n", entity)
+				return api.Allowed
+			}
+		}
+	}
+
+	if entitiesFound {
+		return api.Denied
+	}
 	return api.Undecided
 }
+
+func (r *rule) getEntitySelectors() []api.EndpointSelector {
+	entitiesSelectors := []api.EndpointSelector{}
+
+	for _, r := range r.Egress {
+		for _, entity := range r.ToEntities {
+			selector, ok := api.EntitySelectorMapping[entity]
+			if !ok {
+				continue
```

Comment:
Nit: why not just have:
```
if ok {
    entitiesSelectors = append(entitiesSelectors, selector)
}
```
?

Reply:
thanks, removed these checks altogether as they are validated before